staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -58,16 +61,19 @@ func transformResponseObject(ctx context.Context, scope RequestScope, req *http.
 				scope.err(err, w, req)
 				return
 			}
+			trace.Step("Getting object kind, setting GVK")
 			partial := meta.AsPartialObjectMetadata(m)
 			partial.GetObjectKind().SetGroupVersionKind(metav1beta1.SchemeGroupVersion.WithKind("PartialObjectMetadata"))
 
 			// renegotiate under the internal version
+			trace.Step("Renegotiating internal version")
 			_, info, err := negotiation.NegotiateOutputMediaType(req, metainternalversion.Codecs, &scope)
 			if err != nil {
 				scope.err(err, w, req)
 				return
 			}
 			encoder := metainternalversion.Codecs.EncoderForVersion(info.Serializer, metav1beta1.SchemeGroupVersion)
+			trace.Step(fmt.Sprintf("Serializing response as type %s", info.MediaType))
```

Comment:
I'm 99,9% sure that all the time will spent after this.

So TBH, I would skip the previous steps (as tracing itself has some overhead) and leave just this one...

Reply:
Again trying to get a tight bracket around the "responsewriters.SerializeObject(info.MediaType, encoder, w, req, statusCode, partial)" call.

Reply:
Same point here (sorry for being completely unclear on that).

I completely agree that this Step has value.
I just suggest removing those previous two ones (that should be negligible):
- trace.Step("Getting object kind, setting GVK")
- trace.Step("Renegotiating internal version")

Reply:
Done